iFixit/react-commerce
# Update pnpm to version 8

## Summary
1. Ran `npm install -g pnpm@8` and `pnpm install:all` to update the pnpm version to `v8`.
2. Updated the readme and `pnpm-setup` action.

Note: the vercel deployments seem to be failing because it's using an old build cache: `Restored build cache`.

## QA notes
1. Do a smoke test on all pages: collections (parts, tools), side cart, product pages (regular product and FPP like [rbt](https://www.ifixit.com/products/repair-business-toolkit)), and all other existing pages.
2. Make sure all features are still working, such as adding products to the cart, and searching on collection pages.
3. Do a local setup with pnpm v8 using the following steps:
    - git co update-pnpm
    - `run pnpm -v` and confirm that it's using pnpm v7.
    - then run `pnpm store path` which will return a directory. Then remove that folder usin `rm -rf <some path>`.
    - delete the following folders from your local machine if they exists already: 
         - `react-commerce/node_modules`
         -  frontend: `react-commerce/frontend/node_modules`, `react-commerce/frontend/.next` 
         - backend: `react-commerce/backend/node_modules`, `react-commerce/backend/.cache`, `react-commerce/backend/.tmp`, `react-commerce/backend/dist`
     - then run `npm install -g pnpm@8` 
     - `run pnpm -v` and confirm that it's using pnpm v8.
     - `pnpm install:all`
     - `pnpm run dev` and make sure everything still works.

## On Deploy
Remind everyone to update their `pnpm` version to 8 in their local machine in `dev` channel for react-commerce:
-  `npm install -g pnpm@8`

closes #1685
